### ISSUE TITLE
Effective maskd conv for Guided Anchor?

### ISSUE THREAD

**Issue by kaixinbear** [open-mmlab/mmdetection#2611]
I have tested the time of masked conv and conv,while I find that the general conv performs better than masked conv.

```
import torch
import time
from torch import nn
device = torch.device('cuda')
# device = torch.device("cpu")
gpu_id_used = 1
torch.cuda.set_device(gpu_id_used)
channels_in = 512
channels_out = 512
tensor_w = 256
tensor_h = 128
kernel_size = 3
data_conv_layer = MaskedConv2d(channels_in, channels_out, kernel_size, stride=1, padding=1)
data_conv_layer2 = nn.Conv2d(channels_in, channels_out, kernel_size, stride=1, padding=1)
data_conv_layer.to(device)
data_conv_layer2.to(device)
data = torch.randn((1, channels_in, tensor_h, tensor_w))
data = data.to(device)
repeate_time = 50
data_mask = torch.FloatTensor(tensor_h, tensor_w).uniform_() > 0.7
data_mask = data_mask.unsqueeze(0).to(device)
start_time_val = time.time()
for idx in range(repeate_time):
    data = data_conv_layer(data, data_mask)
total_time_val = time.time() - start_time_val
print('masked_conv input mask : avg time is: {:.6f}'.format(total_time_val/repeate_time))

start_time_val1 = time.time()
for idx in range(repeate_time):
    data = data_conv_layer(data, None)
total_time_val1 = time.time() - start_time_val1
print('masked_conv input None : avg time is: {:.6f}'.format(total_time_val1/repeate_time))

start_time_val2 = time.time()
for idx in range(repeate_time):
    data = data_conv_layer2(data)
total_time_val2 = time.time() - start_time_val2
print('conv avg time is: {:.6f}'.format(total_time_val2/repeate_time))
```

![image](https://user-images.githubusercontent.com/33575107/81037240-87285600-8ed4-11ea-9d6b-35b803522e3e.png)
If this has be proven that masked conv is not better than ordinary conv,is there any necessary
to predict a location probability map in GA-RPN???
Thx for any reply!

**Comment by hellock** [comment]
In our experiments for the paper, masked conv runs a little faster than the normal conv as reported in the paper. Since the normal conv can get optimized during the upgrades of PyTorch and CUDNN, it may be faster now but I am not sure about that. You may try inferring the GARPN model and test the time w/ and w/o masked conv.
